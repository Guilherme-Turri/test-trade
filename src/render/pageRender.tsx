import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};