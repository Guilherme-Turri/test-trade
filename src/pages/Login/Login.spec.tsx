import { pageRender } from "../../render/pageRender";
import { Login } from ".";
import { screen } from '@testing-library/react'

describe('<Login />', () => {
  it('should render on screen', () => {
    const { container } = pageRender(<Login />)
    const logo = screen.getByText('Meu Time')
    expect(logo).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })
})