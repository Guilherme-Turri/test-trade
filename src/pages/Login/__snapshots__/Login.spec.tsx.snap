// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render on screen 1`] = `
.c3 {
  color: #00c555;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c1 .c2 {
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  padding-right: 1rem;
}

.c8 {
  color: #F3F4F5;
  font-weight: bold;
  padding: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  background-image: url(wave.svg);
  background-repeat: no-repeat;
  background-size: cover;
  width: 100%;
  height: 320px;
  z-index: -10000;
}

.c6 {
  outline: none;
  all: unset;
  border: 1px solid #00c555;
  border-radius: 5px;
  width: 100%;
  height: 25px;
  padding: 15px;
  box-sizing: border-box;
  color: #2d5839;
}

.c6::-webkit-input-placeholder {
  color: '#95A5A0'!important;
}

.c6::-moz-placeholder {
  color: '#95A5A0'!important;
}

.c6:-ms-input-placeholder {
  color: '#95A5A0'!important;
}

.c6::placeholder {
  color: '#95A5A0'!important;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  all: unset;
  background: #CACCb1;
  width: 100%;
  border-radius: 5px;
  height: 2rem;
  text-align: center;
  color: #F3F4F5;
  cursor: pointer;
}

.c5 {
  background-color: white;
  box-shadow: 0 2px 10px rgba(0,0,0,0.2);
  border-radius: 5px;
  padding: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 500px;
  height: 250px;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  margin-bottom: 15px;
}

.c4 {
  text-transform: uppercase;
  color: #00c555;
  font-size: 60px;
  font-family: 'Bruno Ace SC',cursive;
}

<div>
  <section
    class="c0"
  >
    <section
      class="c1"
    >
      <a
        class="c2 c3"
        href="https://www.api-football.com/documentation-v3"
        target="_blank"
      >
        NÃ£o tem uma ApiKey? Crie aqui!
      </a>
      <h1
        class="c4"
      >
        Meu Time
      </h1>
    </section>
    <form
      class="c5"
      data-testid="form"
    >
      <input
        autocomplete="new-password"
        class="c6"
        data-testid="Input"
        placeholder="Nome"
        type="text"
        value=""
      />
      <input
        autocomplete="new-password"
        class="c6"
        data-testid="Input"
        placeholder="ApiKey"
        type="password"
        value=""
      />
      <button
        class="c7"
      >
        Enter
      </button>
    </form>
    <footer
      class="c8"
    >
      Guilherme Turri - Trade Technology
    </footer>
  </section>
</div>
`;
